Duplicati/Library/Modules/Builtin/SendMail.cs

```diff
@@ -231,42 +231,10 @@ protected override void SendMessage(string subject, string body)
             List<string> servers = null;
             if (string.IsNullOrEmpty(m_server))
             {
-                var dnslite = new DnsLib.DnsLite();
-                var dnslist = new List<string>();
+                var dnsclient = new LookupClient();
+                var records = dnsclient.Query(toMailDomain, QueryType.MX).Answers.MxRecords();
 
-                //Grab all IPv4 addresses
-                foreach (NetworkInterface networkInterface in NetworkInterface.GetAllNetworkInterfaces())
-                    try
-                    {
-                        foreach (IPAddress dnsAddress in networkInterface.GetIPProperties().DnsAddresses)
-                            if (dnsAddress.AddressFamily == AddressFamily.InterNetwork)
-                                dnslist.Add(dnsAddress.ToString());
-                    }
-                    catch (Exception ex) { Logging.Log.WriteExplicitMessage(LOGTAG, "DNSServerLookupFailure", ex, "Failed to get DNS servers from network interface"); }
-
-                dnslist = dnslist.Distinct().ToList();
-
-                // If we have no DNS servers, try Google and OpenDNS
-                if (dnslist.Count == 0)
-                {
-                    // https://developers.google.com/speed/public-dns/
-                    dnslist.Add("8.8.8.8");
-                    dnslist.Add("8.8.4.4");
-
-                    //http://www.opendns.com/opendns-ip-addresses/
-                    dnslist.Add("208.67.222.222");
-                    dnslist.Add("208.67.220.220");
-                }
-
-                var records = new List<MXRecord>();
-                foreach (var s in dnslist)
-                {
-                    var res = dnslite.getMXRecords(toMailDomain, s);
-                    if (res != null)
-                        records.AddRange(res.OfType<MXRecord>());
-                }
-
-                servers = records.OrderBy(record => record.preference).Select(x => "smtp://" + x.exchange).Distinct().ToList();
+                servers = records.OrderBy(record => record.Preference).Select(x => "smtp://" + x.Exchange).Distinct().ToList();

```

Comment:
I don't have much experience using Duplicati's email options, but on the `master` branch I can get the email reports working simply by using the `send-mail-to` option with a gmail address.  On this branch, I encounter 5 warnings (unfortunately it seems our logging system is broken on this branch), and no email is received.  Are you able to reproduce this?

Reply:
Looks like it works after setting `send-mail-url` to `smtp://alt4.gmail-smtp-in.l.google.com`.  Any idea why that is now required?

Reply:
Found some logs (newest on top):

```
Jun 7, 2021 4:15 PM: Failed to send message: MailKit.Security.SslHandshakeException: An error occurred while attempting to establish an SSL or TLS connection. 
The SSL certificate presented by the server is not trusted by the system for one or more of the following reasons: 
1. The server is using a self-signed certificate which cannot be verified. 
2. The local system is missing a Root or Intermediate certificate needed to verify the server's certificate. 
3. The certificate presented by the server is expired or invalid. 
See https://github.com/jstedfast/MailKit/blob/master/FAQ.md#InvalidSslCertificate for possible solutions. --> System.Security.Authentication.AuthenticationException: The remote certificate was rejected by the provided RemoteCertificateValidationCallback.

Jun 7, 2021 4:15 PM: Whole SMTP communication: Connected to smtp://alt4.gmail-smtp-in.l.google.com.:25/?starttls=when-available S: 220 mx.google.com ESMTP x7si11131319ybj.79 - gsmtp C: EHLO [192.168.143.95] S: 250-mx.google.com at your service, [136.24.158.126] S: 250-SIZE 157286400 S: 250-8BITMIME S: 250-STARTTLS S: 250-ENHANCEDSTATUSCODES S: 250-PIPELINING S: 250-CHUNKING S: 250 SMTPUTF8 C: STARTTLS S: 220 2.0.0 Ready to start TLS
```

I get the same error when setting `send-mail-url` to `smtp://alt4.gmail-smtp-in.l.google.com:25`.  However, when running from the `master` branch and without setting `send-mail-url`, the email is successful (via `smtp://gmail-smtp-in.l.google.com:25/?starttls=when-available`).

Reply:
Maybe relevant?
https://github.com/jstedfast/MailKit/issues/1207

Reply:
Nice find.  It works now if I include
```
client.ServerCertificateValidationCallback = (s, c, h, e) => true;
```
This is [not recommended](https://github.com/jstedfast/MailKit/blob/master/FAQ.md#SslHandshakeException), but the other options don't seem very satisfactory either.

Reply:
In the interest of moving the .NET 5 migration forward, I'll propose that for now we will simply accept any certificate using the following patch:
  ```
diff --git a/Duplicati/Library/Modules/Builtin/SendMail.cs b/Duplicati/Library/Modules/Builtin/SendMail.cs
index 29de81cb2..3970c762c 100644
--- a/Duplicati/Library/Modules/Builtin/SendMail.cs
+++ b/Duplicati/Library/Modules/Builtin/SendMail.cs
@@ -270,6 +270,7 @@ namespace Duplicati.Library.Modules.Builtin
                                 if (uri.Scheme.ToLowerInvariant() == "tls")
                                     uri = new System.Uri("smtp://" + uri.Host + ":" + (uri.Port <= 0 ? 587 : uri.Port) + "/?starttls=always");
 
+                                client.ServerCertificateValidationCallback = (s, c, h, e) => true;
                                 client.Connect(uri);
 
                                 if (!string.IsNullOrEmpty(m_username) && !string.IsNullOrEmpty(m_password))
  ```

I created issue #4552 to consider safer approaches for validating the certificates.

The rest of this pull request looks good to me.  Once we add the above, I think we can merge.

Reply:
Do you want to push that change or do you need me to do it?

Reply:
I tested this again, and it now seems to work without the hack.  Looks like it is now using `smtp://gmail-smtp-in.l.google.com` instead of `smtp://alt4.gmail-smtp-in.l.google.com`.  Not sure why that's the case, but I think we can proceed without the hack for now.